src/crypto-api.ts

```diff
@@ -374,9 +374,22 @@ export interface CryptoApi {
      * This should be called if the client has received the key from another device via secret sharing (gossiping).
      *
      * @param key - the backup decryption key
+     *
+     * @deprecated prefer `version` parameter variant.
      */
     storeSessionBackupPrivateKey(key: Uint8Array): Promise<void>;
 
+    /**
+     * Store the backup decryption key.
+     *
+     * This should be called if the client has received the key from another device via secret sharing (gossiping).
+     *
+     * @param key - the backup decryption key
+     * @param version - the backup version corresponding to this decryption key
+     *
+     */
+    storeSessionBackupPrivateKey(key: Uint8Array, version?: string): Promise<void>;
```

Comment:
shouldn't this be 

```suggestion
    storeSessionBackupPrivateKey(key: Uint8Array, version: string): Promise<void>;
```

?

Reply:
Not sure, I though it was the way to do overloads

Reply:
It's not required for overloads in an `interface`.

When you have overloads in a concrete class the the signature of the *implementation* must be compatible with all overloads (the "implementation signature"). But there is no implementation here.

https://www.typescriptlang.org/docs/handbook/2/functions.html#overload-signatures-and-the-implementation-signature